cloud/scope/machine.go

```diff
@@ -181,6 +182,11 @@ func (m *MachineScope) SetAnnotation(key, value string) {
 	m.GCPMachine.Annotations[key] = value
 }
 
+// SetAddresses sets the addresses field on the GCPMachine.
+func (m *MachineScope) SetAddresses(addressList *[]v1.NodeAddress) {
```

Comment:
Does this need to be a pointer to a slice?

Reply:
Good call, will remove those.